velox/functions/prestosql/ArraySort.cpp

```diff
@@ -374,13 +374,13 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures(
   std::vector<std::shared_ptr<exec::FunctionSignature>> signatures = {
       // array(T) -> array(T)
       exec::FunctionSignatureBuilder()
-          .typeVariable("T")
+          .orderableTypeVariable("T")
           .returnType("array(T)")
           .argumentType("array(T)")
           .build(),
       // array(T), function(T,U), boolean -> array(T)
       exec::FunctionSignatureBuilder()
-          .typeVariable("T")
+          .orderableTypeVariable("T")
```

Comment:
Here, T doesn't have to be orderable, but U has to be.

For example, it should be possible to sort array of maps by cardinality of these maps.

Reply:
For array_sort with a transform lambda, the Presto still checks the T, I describe it in the PR description,
```SQL
presto> SELECT array_sort(ARRAY [map(array['a', 'b', 'c'], array[1, 2, 3])], 
(x, y) -> cardinality(x) < cardinality(y));
...
Expected: array_sort(array(E)) E:orderable, ...
```

Reply:
This sounds like a bug. Let's create an issue in prestodb repo.

Reply:
Got it, I've updated the PR description too.